addons/cargo/functions/fnc_unloadItem.sqf

```diff
@@ -47,14 +47,17 @@ private _space = [_vehicle] call FUNC(getCargoSpaceLeft);
 private _itemSize = [_item] call FUNC(getSizeItem);
 _vehicle setVariable [QGVAR(space), (_space + _itemSize), true];
 
-if (_item isEqualType objNull) then {
+private _itemObject = if (_item isEqualType objNull) then {
     detach _item;
     // hideObjectGlobal must be executed before setPos to ensure light objects are rendered correctly
     // do both on server to ensure they are executed in the correct order
     [QGVAR(serverUnload), [_item, _emptyPosAGL]] call CBA_fnc_serverEvent;
+    _item;
 } else {
     private _newItem = createVehicle [_item, _emptyPosAGL, [], 0, "NONE"];
     _newItem setPosASL (AGLtoASL _emptyPosAGL);
+    _newItem;
 };
-
+// Invoke listenable event
+["ace_cargoUnloaded", [_itemObject, _vehicle, "unload"]] call CBA_fnc_globalEvent;
 true
```

Comment:
1. What is an "itemObject"? Why name it "itemX" when it explictily is not an "item" (i.e. classname), but an "object"?
2. Do not use the rv on control structures.

```suggestion
private _object = _item;

if (_item isEqualType objNull) then {
    detach _item;
    // hideObjectGlobal must be executed before setPos to ensure light objects are rendered correctly
    // do both on server to ensure they are executed in the correct order
    [QGVAR(serverUnload), [_item, _emptyPosAGL]] call CBA_fnc_serverEvent;
} else {
    _object = createVehicle [_item, _emptyPosAGL, [], 0, "NONE"];
    _object setPosASL (AGLtoASL _emptyPosAGL);
};

// Invoke listenable event
["ace_cargoUnloaded", [_object, _vehicle, "unload"]] call CBA_fnc_globalEvent;
true
```

Reply:
1.1 _itemObject is a place for a reference to an object that is being unloaded, that was either just created or moved into position if it already existed.
1.2 I called the variable the same way it was called in paradropItem because it does exactly the same thing.
2. ok, should I fix paradropItem to follow this rule?

Reply:
That would be sweet, because it sounds like my criticisms extend to the old code as well. Although I would be fine with just not writing new non-sense. Perhaps a second pull could be made to sanitize the other code, or maybe it can be made to fit into this one. Up to you if you want to do that.

Reply:
all done